developers/advanced/null-value-support.md

```diff
@@ -37,7 +40,35 @@ Remember that in the JSON used as table configuration, `defaultNullValue` must a
 If the column type is not String, Pinot will convert that value to the column type automatically.
 {% endhint %}
 
-## Store nulls at ingestion time
+### Enable basic null support for the single-stage query engine
+
+To enable basic null support (`IS NULL` and `IS NOT NULL`) and generate the null index, in the Table index configuration ([tableIndexConfig](https://docs.pinot.apache.org/configuration-reference/table#tableindexconfig-1)), set `nullHandlingEnabled=true`.
```

Comment:
This is misleading. Both table index config (`nullHandlingEnabled`) and column level nullability (set in the schema) enable basic null handling.

I don't get why you propose to separate multi and single stage here. It is just making it even more confusing.

It is also leaving the tendency I followed in my version where I called this _null handling storage_ instead of _basic null support_, which is very important because it is more precise and less confusing. Otherwise we always use null handling and differentiate it in very sutile ways, which is IMHO super confusing.

What I wanted to say in my version is:
1. You have to enable null handling storage